nri-postgresql: use optional prefix for all env vars in decoderWithPrefix

zwilias opened the pull request:
- Allow running tests locally
- pg: Use optional prefix for decoderWithPrefix

jwoudenberg-nri commented on nri-log-explorer/src/Main.hs:
```diff
@@ -1238,50 +1235,26 @@ updateTime withTime = do
+      |> andThen
+        ( Streams.atEndOfInput <| do
+            Control.Concurrent.threadDelay 100_000 {- 100 ms -}
+            streamLines withLine handle
```
💭 I think it's possible that by the time we reach this point we're in the middle of a JSON line (because writing lines of JSON isn't atomic). That would mean that whatever JSON line is being written as we reach this point is going to be dropped.

Also, not super happy with the timeout here. That seems less nice then what we had before.

stoeffel replied:
Doesn't this reuse the handle and that stays at the position it was?

stoeffel replied:
I think it's like it was before. Keeping the handle open

zwilias committed:
Allow running tests locally

zwilias committed:
pg: Use optional prefix for decoderWithPrefix

This changes behaviour in two ways:
- all parts will use the prefix, not just the connection settings
- if the prefixed env variable is not found, it will first fall back to
  the unprefixed env variable, before falling back to the defined
  default.

I expect this to be fine.

asterite reviewed (approved):
![](https://media0.giphy.com/media/m6OYxK7CMwES9CzDA7/200w.webp?cid=ecf05e47msd5qg6rwcwyt55yc6tq8izbao0k7k58rouked62&ep=v1_gifs_search&rid=200w.webp&ct=g)

@NoRedInkBot r+
